### PR TITLE
Remove coupling between ConnectDialog and selected signal

### LINKED ISSUE #43219: Method created from editing connection doesn't include arguments
**Godot version:**

3.2.3

**Steps to reproduce:**
1. Create a node (e.g. AnimationPlayer) and add a script
2. Connect some signal that has arguments
3. A method with the arguments will be created
4. Now edit the connection and change the target method
5. New method will be generated, but without arguments

Cr3w9iPWQm

### PR DESCRIPTION
**Pull request by MinusKube**
This removes the coupling between the ConnectDialog and the selected signal tree item from the connections dock, fixing bugs like https://github.com/godotengine/godot/issues/63710#issuecomment-1399544863.

This also fixes a bug which caused the arguments of the newly created script function to not include the default signal args, when editing an existing signal connection (discovered this bug while fixing the first one, and I don't think there's an open issue for it).

Finally, this fixes a third bug which caused the undo to fail when using "Disconnect" on a signal connection or "Disconnect All" on a signal, because the args given to `add_undo_method` were wrong (also discovered this bug while fixing the first one, and I don't think there's an open issue for it either).

*Bugsquad edit:* Fixes https://github.com/godotengine/godot/issues/43219

### CODE REVIEW

**Comment by MinusKube** [comment]
Also, I saw there's #71874 which fixes the same first bug, and actually, I think both PR should be merged, because their PR fixes the selection problem while mine fixes the underlying coupling problem. Having the both issues fixed would make the system more robust.

File: editor/connections_dialog.cpp [hunk 1]
```diff
@@ -809,22 +796,30 @@ void ConnectionsDock::_open_connection_dialog(TreeItem &p_item) {
...
+	Node *src = Object::cast_to<Node>(cd.source);
+	Node *dst = Object::cast_to<Node>(cd.target);
 
 	if (src && dst) {
-		const String &signal_name_ref = p_cd.signal;
-		connect_dialog->set_title(TTR("Edit Connection:") + p_cd.signal);
+		const String &signal_name_ref = cd.signal;
+		PackedStringArray signal_args = signal_item->get_metadata(0).operator Dictionary()["args"];
+
+		connect_dialog->set_title(TTR("Edit Connection:") + cd.signal);
```
Comment (KoBeWi) [inline editor/connections_dialog.cpp:820]:
```suggestion
		connect_dialog->set_title(TTR("Edit Connection:") + " " + cd.signal);
```
This wasn't here originally, but makes it look better. Alternatively:
```suggestion
		connect_dialog->set_title(vformat(TTR("Edit Connection: %s", cd.signal));
```

Comment (MinusKube) [reply editor/connections_dialog.cpp]: Done!

**Review by KoBeWi — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**